extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/entitymanager/TransactionScopedEntityManager.java

```diff
@@ -34,7 +34,7 @@
     private final TransactionSynchronizationRegistry tsr;
     private final EntityManagerFactory emf;
     private final String unitName;
-    private static final Object transactionKey = new Object();
+    private final Object unitEntityManagerInTxKey = new Object();
```

Comment:
I have a slightly different fix where I build a String key from the class simple name + ":" + the unit name in the constructor and use that.

It also fixes the issue and I think that's more in line with what we used to do.

If it makes sense to you, could you adjust your patch?

Reply:
yes, give me a sec ...